<?xml version="1.0" encoding="UTF-8" standalone="yes"?>

<testsuite name="Store Object from List">
	<test name="StoreObjectFromList - Storing one object ">
		<automationValue>High</automationValue>
		<teststeps>
			<storeDefaultObject contextKey="verifyList">
				<list name="default">
					<string name="Test">Test value in context</string>
				</list>
			</storeDefaultObject>
			<storeObjectFromList list="verifyList" object="newItem"
				index="1" />
			<verifyObjectInContext actualKey="newItem"
				expectedKey="testKey" testKey="Test value in context" />
		</teststeps>
	</test>



	<test name="StoreObjectFromList - Storing two object ">
		<automationValue>High</automationValue>
		<teststeps>
			<storeDefaultObject contextKey="verifyList">
				<list name="default">
					<string name="Value1">Value1</string>
					<string name="Value2">Value2</string>
				</list>
			</storeDefaultObject>
			<storeObjectFromList list="verifyList" object="newItem1"
				index="1" />
			<verifyObjectInContext actualKey="newItem1"
				expectedKey="testKey" testKey="Value1" />

			<storeObjectFromList list="verifyList" object="newItem2"
				index="2" />
			<verifyObjectInContext actualKey="newItem1"
				expectedKey="testKey" testKey="Value1" />
			<verifyObjectInContext actualKey="newItem2"
				expectedKey="testKey" testKey="Value2" />

		</teststeps>
	</test>



	<test name="StoreObjectFromList - Exception">
		<automationValue>High</automationValue>
		<teststeps>
			<storeDefaultObject contextKey="verifyList">
				<list name="default">
					<string name="Value1">Value1</string>
				</list>
			</storeDefaultObject>
			<verifyException exceptionType="Exception">
				<storeObjectFromList list="verifyList" object="newItem1"
					index="0" />
			</verifyException>

		</teststeps>
	</test>

</testsuite>
